mlflow/gateway/providers/gemini.py

```diff
@@ -1,12 +1,246 @@
+                    status_code=422, detail=f"Invalid parameter {k2}. Use {k1} instead."
+                )
+
+        if "top_p" in payload and payload["top_p"] > 1:
+            raise AIGatewayException(
+                status_code=422, detail="top_p should be less than or equal to 1"
+            )
+
+        payload = rename_payload_keys(payload, GENERATION_CONFIG_KEY_MAPPING)
+
+        contents, system_message = [], None
+        for message in payload["messages"]:
+            role = message["role"]
+
+            if role == "assistant":
+                role = "model"
+            elif role == "system":
+                if system_message is None:
+                    system_message = {"parts": []}
+                system_message["parts"].append({"text": message["content"]})
+                continue
+
+            contents.append({"role": role, "parts": [{"text": message["content"]}]})
+
+        gemini_payload = {"contents": contents}
+
+        if system_message:
+            gemini_payload["system_instruction"] = system_message
+
+        generation_config = {k: v for k, v in payload.items() if k in GENERATION_CONFIGS}
+
+        if generation_config:
+            gemini_payload["generationConfig"] = generation_config
+
+        return gemini_payload
+
+    @classmethod
+    def model_to_chat(cls, resp, config):
+        # Documentation: https://ai.google.dev/api/generate-content
+        #
+        # Example Response:
+        # {
+        #   "candidates": [
+        #     {
+        #       "content": {
+        #         "parts": [
+        #           {
+        #             "text": "Blue is often seen as a calming and soothing color."
+        #           }
+        #         ]
+        #       },
+        #       "finishReason": "stop"
+        #     }
+        #   ],
+        #   "usageMetadata": {
+        #     "promptTokenCount": 10,
+        #     "candidatesTokenCount": 10,
+        #     "totalTokenCount": 20
+        #   }
+        # }
+        choices = []
+        for idx, candidate in enumerate(resp.get("candidates", [])):
+            content = ""
+            if parts := candidate.get("content", {}).get("parts", None):
+                content = parts[0].get("text", None)
+            if not content:
+                continue
+
+            finish_reason = candidate.get("finishReason", "stop")
+            if finish_reason == "MAX_TOKENS":
+                finish_reason = "length"
+
+            choices.append(
+                chat.Choice(
+                    index=idx,
+                    message=chat.ResponseMessage(
+                        role="assistant",
+                        content=content,
+                    ),
+                    finish_reason=finish_reason,
+                )
+            )
+
+        usage_metadata = resp.get("usageMetadata", {})
+
+        return chat.ResponsePayload(
+            id=resp.get("promptFeedback", {}).get("promptId", f"gemini-chat-{int(time.time())}"),
```

Comment:
Does `promptId` field exist in the promptFeedback object?
https://ai.google.dev/api/generate-content#PromptFeedback